src/Headers.php

```diff
@@ -227,13 +218,19 @@ public function addHeaderLine($headerFieldNameOrLine, $fieldValue = null)
         }
 
         if ($fieldValue === null) {
-            $this->addHeader(Header\GenericHeader::fromString($headerFieldNameOrLine));
+            list($fieldName, $fieldValue) = Header\GenericHeader::splitHeaderLine($headerFieldNameOrLine);
+            $norm = $this->normalizeFieldName($fieldName);
+            $this->headersKeys[] = $norm;
+            $this->headers[] = $headerFieldNameOrLine;
```

Comment:
addHeader is forcing the encoding, which break many mails
(i've parsed more than 100000 mails (only the headers) in my first PR)

for the bigger view, it's to fix the lazyloading, don't need to parse before using the header
(some explanation in the commit also)